jni/instruments/oscillatorproperties.cpp

```diff
@@ -54,12 +54,23 @@ void OscillatorProperties::setWaveform( int aWaveform )
         delete waveTable;
 
     if ( aWaveform == WaveForms::TABLE ) {
-        WaveTable* tableFromPool = TablePool::getTable( aWaveform );
-
-        if ( tableFromPool != 0 )
-            waveTable = tableFromPool->clone();
-        else
-            aWaveform = WaveForms::SINE;
+        // use setCustomWaveform instead
+        return;
     }
     _waveform = aWaveform;
 }
+
+void OscillatorProperties::setCustomWaveform( std::string waveformId )
+{
```

Comment:
When using a custom waveform, you specify the identifier under which it has been registered inside the tablepool (see tablepool.h). If it hasn't been registered, default to sine.